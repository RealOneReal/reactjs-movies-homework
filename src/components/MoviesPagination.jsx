import { Pagination, PaginationItem } from "@mui/material";

const MoviesPagination = (props) => {
  // todo: realize pagination after connecting Redux
  const { total_pages, page } = props;
  return (
    <Pagination
      count={total_pages}
      page={page}
      hideNextButton={true}
      hidePrevButton={true}
      color="primary"
      renderItem={(item) => (
        <PaginationItem
          component="button"
          sx={{ color: "primary.dark" }}
          {...item}
        />
      )}
    />
  );
};

export default MoviesPagination;
