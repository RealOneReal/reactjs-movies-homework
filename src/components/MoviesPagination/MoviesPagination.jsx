import { Pagination, PaginationItem } from "@mui/material";

const MoviesPagination = ({ total_pages, page }) => {
  // todo: realize pagination after connecting Redux
  return (
    <Pagination
      count={total_pages}
      page={page}
      hideNextButton={true}
      hidePrevButton={true}
      color="primary"
      renderItem={(item) => (
        <PaginationItem
          component="button"
          sx={{ color: "primary.dark" }}
          {...item}
        />
      )}
    />
  );
};

export default MoviesPagination;
