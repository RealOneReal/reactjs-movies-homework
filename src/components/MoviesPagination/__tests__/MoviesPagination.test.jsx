import renderer from "react-test-renderer";
import MoviesPagination from "../MoviesPagination";

describe("MoviesPagination", () => {
  it("should be render", () => {
    const tree = renderer.create(<MoviesPagination />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
