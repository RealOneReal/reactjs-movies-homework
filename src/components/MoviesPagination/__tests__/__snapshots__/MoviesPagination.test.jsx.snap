// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesPagination should be render 1`] = `
<nav
  aria-label="pagination navigation"
  className="MuiPagination-root MuiPagination-text css-1oj2twp-MuiPagination-root"
>
  <ul
    className="MuiPagination-ul css-wjh20t-MuiPagination-ul"
  >
    <li>
      <button
        aria-current="true"
        aria-label="page 1"
        className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-textPrimary Mui-selected MuiPaginationItem-page css-7oxwxk-MuiButtonBase-root-MuiPaginationItem-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        1
      </button>
    </li>
  </ul>
</nav>
`;
